infrasec/aws/vpcs.md

```diff
@@ -0,0 +1,82 @@
+When in doubt, use a `/16`.
+
+Note that each region has multiple [Availability Zones](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones) and the ip space will be subdivided into subnets across these availabilty zones.
+
+The following are objects associated with VPCs that can be configured.
+
+### Subnets
+
+A subnet is a logical subdivision of an IP network.
+Generally, you will create a pair of public and private subnets in each availability zone for the region.
+A good default CIDR range for each subnet is either a `/22` or `/24`.
+
+Creating additional subnets will create additional complexity in route tables.
+If your project requires you to isolate resources from one another, use Security groups.
+
+### Route tables
+
+Route tables define the network routing in the VPC.
+You can route specific ip address spaces to separate subnets or to gateways.
+
+### Network ACLs
+
+Network ACLs are like firewall rules applied to specific subnets in a VPC.
+You can define inbound and outbound network traffic across ports for specific IP addresses.
+We prefer to use Security Groups over Network ACLs as they can be applied to specific resources in a VPC.
+
+### Security Groups
+
+In a traditional corporate network these would be equivalent to firewall rules.
+Security Groups allow you to define inbound and outbound network traffic across ports that than be specific to IP addresses OR other security groups.
+Because you can define security group access using security group names, we find it easier to configure intended access to resources than Network ACLs.
+
+### Gateways
+
+For all public subnets in a VPC, use a singlle internet gateway for default routes.
+This means they can route to the public internet.
+
+For each private subnet, create a NAT gateway with an EIP.
+You do not need a NAT instance.
+
+### VPC endpoints
+
+VPC endpoints are AWS provided and defined endpoints for AWS services.
+You may want to use this to close up your network so you have no public traffic.
+
+There are two types of VPC endpoints, Gateway endpoints and Interface endpoints.
+
+Interface endpoints are a deployed elastic network interface with a private IP address connected to your VPC.
+Note: In our internal test configuration we weren't sure it was worth the cost to just spin up one of everything.
+
+Gateway endpoints are existing endpoints that require configuration in the route tables.
+Available Gateway endpoints are: AWS S3 and AWS DynamoDB endpoints in your VPC and you should just deploy those.
+
+The full list of supported endpoints is [here](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-endpoints.html).
+
+### Flow logs
+
+VPC Flow logs allow you to monitor all network traffic in your VPC.
+Unless you have a plan to ingest, manage, view, and monitor network flow logs we do not recommend you turn these on.
```

Comment:
So here's a question, that I don't 100% know the answer. AWS GuardDuty relies on VPC flow logs to be able to detect things like SSH brute force attempts. What I don't know if we need to turn flow logs on for that to work or if AWS somehow behind the scenes is able to use flow logs even if we leave them off.

Reply:
/me lies facedown on the floor

Maybe for now I'll throw in this note so folks know we don't know that yet... and maybe put a task in our backlog to research that.

Reply:
https://www.pivotaltracker.com/story/show/174102611 added this to look into it